lib/MusicBrainz/Server/Controller/Artist.pm

```diff
@@ -360,6 +360,9 @@ sub recordings : Chained('load')
     my $standalone_only;
     my $video_only;
 
+    my $has_standalone = scalar $c->model('Recording')->find_standalone($artist->id, 1);
+    my $has_videos = scalar $c->model('Recording')->find_video($artist->id, 1);
```

Comment:
We should add custom methods to the data layer for these, like `has_standalone` and `has_video`. The first query resolves to e.g.

```SQL
WITH x AS (
    SELECT DISTINCT recording.id, recording.gid, recording.name,
        recording.artist_credit AS artist_credit_id,
        recording.length, recording.comment, recording.video,
        recording.edits_pending, recording.last_updated,
        recording.name COLLATE musicbrainz
      FROM recording
    LEFT JOIN track t ON t.recording = recording.id
      JOIN artist_credit_name acn
        ON acn.artist_credit = recording.artist_credit
     WHERE t.id IS NULL
       AND acn.artist = 303
    ORDER BY recording.name COLLATE musicbrainz
)
SELECT x.*, c.count AS total_row_count
FROM x, (SELECT count(*) from x) c
LIMIT 1;
```

for artist = 303, which takes around 100 ms on pink. With a more optimized query, it only takes ~25 ms (still quite slow, but 4X faster):

```SQL
SELECT EXISTS (
    SELECT 1
      FROM recording
      JOIN artist_credit_name acn
        ON acn.artist_credit = recording.artist_credit
     WHERE acn.artist = 303
       AND NOT EXISTS (SELECT 1 FROM track WHERE track.recording = recording.id)
);
```

Reply:
I think it should be fine now. I thought about using these to avoid running the queries in     `if ($c->req->query_params->{standalone}) ` and whatnot, but a pager is required not to break the React display. Any suggestions for that, or should I just leave it be? :)

Reply:
One thing you could do is return `([], 0)` from the `_load_paged` sub if those vars are set.